src/pages/MuteerOverview/MuteerOverview.tsx

```diff
@@ -0,0 +1,398 @@
+
+/**
+ * A component to display all the objects from a specific dimension
+ * @param {Object} dimensieConstants - Contains the variables of the dimension
+ */
+
+type PossiblePolicyTypes =
+    | AmbitiesRead
+    | BelangenRead
+    | BeleidskeuzesRead
+    | BeleidsregelsRead
+    | BeleidsprestatiesRead
+    | BeleidsmodulesRead
+    | BeleidsdoelenRead
+    | MaatregelenRead
+    | ThemasRead
+    | VerordeningenRead
+
+interface MuteerOverviewProps {
+    dimensieConstants: filteredDimensieConstants
+    hideAddObject?: boolean
+}
+
+const MuteerOverview = ({
+    dimensieConstants,
+    hideAddObject,
+}: MuteerOverviewProps) => {
+    const { user } = useAuth()
+
+    const [filterQuery, setFilterQuery] = useState('')
+    const [isLoading, setIsLoading] = useState(true)
+    const [ascending, setAscending] = useState(true)
+    const [policyObjects, setPolicyObjects] = useState<PossiblePolicyTypes[]>(
+        []
+    )
+
+    const navigate = useNavigate()
+
+    const titleSingular = dimensieConstants.TITLE_SINGULAR
+    const titlePlural = dimensieConstants.TITLE_PLURAL
+    const createNewSlug = dimensieConstants.SLUG_CREATE_NEW
+    const overviewSlug = dimensieConstants.SLUG_OVERVIEW
+
+    const useGetLineage = getFetcherForType(titleSingular)
+    const { data: policyObjectsFromAPI } = useGetLineage()
+
+    /**
+     * When the component mounts, fetch the objects from the API and prepare state:
+     * 1. Sort based on ascending or descending
+     * 2. Filter out archived objects
+     * 3. Filter based on filter Query value
+     * 4. Set in state
+     */
+    useEffect(() => {
+        if (!policyObjectsFromAPI) return
+
+        const sortedAndFilteredPolicyObjects = policyObjectsFromAPI
+            .sort((a, b) =>
+                ascending
+                    ? a.Titel! > b.Titel!
+                        ? 1
+                        : -1
+                    : a.Titel! < b.Titel!
+                    ? 1
+                    : -1
+            )
+            .filter(
+                policy =>
+                    policy.Eind_Geldigheid &&
+                    isValid(new Date(policy.Eind_Geldigheid)) &&
+                    isBefore(new Date(), new Date(policy.Eind_Geldigheid))
+            )
+            .filter(policyObject =>
+                policyObject.Titel!.toLowerCase().includes(filterQuery)
+            )
+
+        setPolicyObjects(sortedAndFilteredPolicyObjects)
+        setIsLoading(false)
+    }, [policyObjectsFromAPI, ascending, filterQuery])
+
+    /** Check if the user is authenticated for the page */
+    useEffect(() => {
+        if (!user) return
+
+        const isBeleidsmodulePage =
+            dimensieConstants.TITLE_SINGULAR === 'Beleidsmodule'
+
+        /** The user is always authenticated for the beleidsmodule page */
+        if (isBeleidsmodulePage) return
+
+        /** Check if user is authenticated */
+        const userRole = user.Rol
+        const userHasAuthenticatedRole =
```

Comment:
@Aidenbuis I think I've seen some kind of authentication check before, so also for this part it would be nice to have a separate utility.

Reply:
See 3b427824af614340f24d7ca5d75ee6fe07e2d08c